### ISSUE TITLE
Scroll history is missing lines

### ISSUE THREAD

**Issue by chtenb** [wez/wezterm#1923]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

wezterm 20220421-183827-26500c67

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Let's print a list of numbers

![image](https://user-images.githubusercontent.com/1030961/165307660-13652010-9bab-4052-b66d-514f146599e0.png)

Then scrolling up

![image](https://user-images.githubusercontent.com/1030961/165307718-66fe2b1a-016f-488c-b3f2-2a2e66dba3c3.png)

there appear to be missing lines right at the edge of the screen

### Configuration

no config

### Logs

15:15:09.359 INFO wezterm_client::discovery::windows > published gui path as gui-sock-23932
15:15:09.419 WARN window::os::windows::window > EGL init failed Config says to avoid EGL, fall back to WGL
15:15:09.687 INFO wezterm_gui::termwindow > OpenGL initialized! Quadro M2200/PCIe/SSE2 4.5.0 NVIDIA 472.91 is_context_loss_possible=true wezterm version: 20220421-183827-26500c67

**Comment by chtenb** [comment]
Is there any chance we'll get a fix for this in the not-too-distant future? It's a pretty crucial bug as I sometimes miss lines in command output, which obviously is something that induces mistakes and slowdowns.

**Comment by wez** [comment]
I think it may be due to:
* https://github.com/microsoft/terminal/issues/8698

as this issue doesn't occur on other platforms.

Are you able to disable coloring of the output to see if the issue goes away? If so, then it adds weight to that theory.

**Comment by wez** [comment]
Other things to look at to try to isolate this:
* Disabling shell integration escape sequences if you're using those
* Otherwise: Simplifying your prompt printing setup. (these may have limited utility if it is the command output that is confusing things)
* Is it possible to try using `wezterm ssh` to connect to the same (presumably local?) host and see if the issue persists? (that removes ConPTY from the wezterm side, but depending on your ssh server, there may be a separate instance on the server side)

It may also be useful to capture a session recording:

* Launch wezterm.  If possible, please use the default terminal size of 80x24 cells as it helps to keep things smaller and easier to manage.
* Inside that terminal run `wezterm record` to start a recording session.
* Run through your reproduction steps
* Then type `exit`
* You should see a message like:

```
*** Finished recording to /var/tmp/wezterm-recording-sF6B3u.cast.txt
```

* Attach the file that it produced to this issue.

The file is an asciicast (compatible with https://asciinema.org/) and can also be replayed using `wezterm replay`.

The terminal recording allows me to replicate what is being sent to the terminal without requiring me to install the same applications as you and replicate your configuration for everything.

**Comment by chtenb** [comment]
[wezterm-recording-vUPbou.cast.txt](https://github.com/wez/wezterm/files/10331082/wezterm-recording-vUPbou.cast.txt)

Here is a cast of me reproducing the issue. 

I noticed however when replaying it with `wezterm replay` that I couldn't see the scroll actions being performed. But the issue is ONLY ever visible when I scroll up and notice that lines are missing there. Even though I have seen the missing lines pass by when they were still visible in the default viewport.

**Comment by chtenb** [comment]
Okay, I was sure I tried reproducing this on other terminals without success before I created the issue here. But I just reproduced it on VSCode's integrated terminal using the same method. This adds more weight to your theory of it being an upstream issue.

What I don't understand though, is that those lines only disappear once they fall off the default viewport and you scroll up. I have limited knowledge about how terminal emulators work but I always thought the history/scroll feature was a emulator-specific implementation.

**Comment by chtenb** [comment]
> Simplifying your prompt printing setup.

Good call. I removed my starship configuration and after that I couldn't reproduce the issue anymore. Neither in VSCode and wezterm. I wonder what it is about starship that throws conpty off.

**Comment by chtenb** [comment]
https://github.com/starship/starship/issues/4635

**Comment by chtenb** [comment]
Is conpty also used when using WSL? I can reproduce the issue both on native windows and WSL.

**Comment by chtenb** [comment]
So there is a workaround posted in the other thread which works for me. Feel free to close this issue if you want. Other people with the same issue can follow the instructions in the linked thread.

**Comment by quantonganh** [comment]
- macOS 12.3
- wezterm 20231112-171650-721fbdf5

To reproduce:
- Download the nightly build: https://wezfurlong.org/wezterm/install/macos.html
- Unzip, run and skip loading config `wezterm -n`
- Run a command to print something:

```
$ for i in (seq 1 30); echo $i $i; end
```

- Clear screen by pressing control-L or typing `clear`
- Re-run the above command
- Scroll up

![Screen_Shot_2023-11-17_at_10_07_39](https://github.com/wez/wezterm/assets/1568504/7359a918-7cc9-4231-9dde-fcdf5e9340d8)

History is not missing if we do not clear the screen.

**Comment by quantonganh** [comment]
To anyone who want to keep scrollback buffer after clear: https://github.com/fish-shell/fish-shell/issues/2855#issuecomment-201377349

**Comment by lkusiani** [comment]
- arch linux, wayland
- wezterm 20250604-060157-5106c8c1
same issue here:
`seq 1 50` generates 50 lines of output, after pressing ctrl-l and scrolling up, only first couple of lines are shown.
